ConsentViewController/Classes/SourcePointClient/Adapters.swift

```diff
@@ -255,6 +257,48 @@ extension SPMobileCore.USNatConsent {
     }
 }
 
+extension SPMobileCore.PreferencesConsent.PreferencesSubType? {
+    func toNative() -> SPPreferencesConsent.PreferencesSubType {
+        switch self {
+        case .aipolicy: return .AIPolicy
+        case .legalpolicy: return .LegalPolicy
+        case .privacypolicy: return .PrivacyPolicy
+        case .termsandconditions: return .TermsAndConditions
+        case .termsofsale: return .TermsOfSale
+        case .unknown: return .Unknown
+        default: return .Unknown
+        }
+    }
+}
+
+extension SPMobileCore.PreferencesConsent.PreferencesStatusPreferencesChannels {
+    func toNative() -> SPPreferencesConsent.PreferencesChannels { return .init(channelId: Int(self.channelId), status: self.status) }
+}
+
+extension SPMobileCore.PreferencesConsent.PreferencesStatus {
+    func toNative() -> SPPreferencesConsent.PreferencesStatus {
+        return .init(
+            categoryId: Int(self.categoryId),
+            channels: self.channels?.map { $0.toNative() },
+            changed: self.changed?.boolValue,
+            dateConsented: SPDate(string: self.dateConsented?.instantToString() ?? SPDate.now().originalDateString),
+            subType: self.subType.toNative()
+        )
+    }
+}
+
+extension SPMobileCore.PreferencesConsent {
+    func toNative() -> SPPreferencesConsent {
+        return .init(
+            dateCreated: SPDate(string: self.dateCreated?.instantToString() ?? SPDate.now().originalDateString),
+            messageId: String(Int(truncating: self.messageId ?? 0)),
```

Comment:
are you sure `messageId` shouldn't be optional? it feels strange having to pass a default value of `0`  to it. I'm concerned this might affect reporting.

Reply:
@andresilveirah `messageId` in Swift is a `String?` and  `messageId` in Kotlin is an `Int?`. 
Casting Kotlin's optional `Int?` into Swift's `Int` is tricky and requires a `?? 0` fallback, as Swift's `Int` cast does not want null.
The best we can do here is
> messageId: self.messageId != nil ? String(Int(truncating: self.messageId ?? 0)) : nil,